FileSystem/FileMapperMocker.js

```diff
@@ -10,28 +25,490 @@ var USE_FILEMAPPER_MOCKER = true;
+ * replaced with complete implementation. This mocker implements all the
+ * functionality as per the design of the FileMapper. The primary difference is
+ * that there is no config file, instead there is a hardcoded base directory.
+ * All file paths have to contain this base directory as prefix. The mocker does
+ * NOT do the following things 1) Check MimeType, Permissions 2) Check for
+ * duplicate files 3) Throw errors as per design (because we are not doing
+ * checks) 4) Prompt user in case file is not found
+ */
+var VBAFileMapperMocker2 = {
+  baseDirectory: 'c:\\user\\desktop',
+  getFileId: getFileId,
+  getFolderId: getFolderId,
+  hasFile: hasFile,
+  hasFolder: hasFolder,
+  deleteFile: deleteFile2,
+  deleteFolder: deleteFolder,
+  createFile: createFile2,
+  createFolder: createFolder,
+  moveFile: moveFile,
+  moveFolder: moveFolder,
+  copyFile: copyFile,
+  copyFolder: copyFolder,
+  findFilesByPattern: findFilesByPattern,
+  findFoldersByPattern: findFoldersByPattern,
+  getRelativePathSplit: getRelativePathSplit,
+  getBaseDriveFolder: getBaseDriveFolder,
+};
+
+var FileMapper = USE_FILEMAPPER_MOCKER ? VBAFileMapperMocker2 : VBAFileMapper;
+
+/**
+ * Get File API
+ * Gets the file id of a local file
+ * @param {string} localPath Local filepath of the file
+ * @return {string} File id of the local file
+ */
+function getFileId(localPath) {
+  // Get platform agnostic relative path split
+  var pathSplit = this.getRelativePathSplit(localPath);
+  // Get base directory
+  var root = this.getBaseDriveFolder();
+  // Find folder containing local file
+  for (var i = 0; i < pathSplit.length - 1; i++) {
+    var folderIterator = root.getFoldersByName(pathSplit[i]);
+    if (!folderIterator.hasNext()) {
+      throw new Error('File not found');
+    }
+    root = folderIterator.next();
+  }
+  // Obtain the file
+  var fileIterator = root.getFilesByName(pathSplit[pathSplit.length - 1]);
+  if (!fileIterator.hasNext()) {
+    throw new Error('File not found');
+  }
+  // Return file id
+  return fileIterator.next().getId();
+}
+
+/**
+ * Get Folder API
+ * Gets the folder id of a local folder
+ * @param {string} localPath Local filepath of the folder
+ * @return {string} Folder id of the local folder
+ */
+function getFolderId(localPath) {
+  // Get platform agnostic relative path split
+  var pathSplit = this.getRelativePathSplit(localPath);
+  // Get base directory
+  var root = this.getBaseDriveFolder();
+  // Find the directory
+  for (var i = 0; i < pathSplit.length; i++) {
+    var folderIterator = root.getFoldersByName(pathSplit[i]);
+    if (!folderIterator.hasNext()) {
+      throw new Error('Folder not found');
+    }
+    root = folderIterator.next();
+  }
+  return root.getId();
+}
+
+/**
+ * Has File API
+ * Checks if the local file exists
+ * @param {string} localPath Local filepath of the file
+ * @return {boolean} true if the local file exists
+ */
+function hasFile(localPath) {
```

Comment:
This method is almost same as getFileId, may be you can make a helper that returns  the id or null and use that in both the methods, (in the above method you can check for null and throw error), similarly for the folder ones

Reply:
Fixed: Refactored to using getFileUtil()

Reply:
Fixed